tests/karma.conf.js

```diff
@@ -0,0 +1,37 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+module.exports = function (config) {
+  config.set({
+    basePath: '',
+    frameworks: ['jasmine', '@angular-devkit/build-angular'],
+    plugins: [
+      require('karma-jasmine'),
+      require('karma-chrome-launcher'),
+      require('karma-firefox-launcher'),
+      require('karma-spec-reporter'),
+      require('karma-coverage-istanbul-reporter'),
+      require('@angular-devkit/build-angular/plugins/karma')
+    ],
+    client: {
+      clearContext: false // leave Jasmine Spec Runner output visible in browser
+    },
+    coverageIstanbulReporter: {
+      dir: require('path').join(__dirname, './coverage/CATcher'),
+      reports: ['html', 'lcovonly', 'text-summary'],
+      fixWebpackSourcePaths: true
+    },
+    reporters: ['spec'],
+    specReporter: {
+      maxLogLines: 5, // limit number of lines per test
+      suppressErrorSummary: true,
+      suppressPassed: true,
+      showSpecTiming: false,
+    },
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    browsers: ['ChromeHeadless', 'FirefoxHeadless'],
```

Comment:
Should we add Safari to the list of browsers? Since its not chromium based and also what some mac users use.

Reply:
This setup requires the browsers to be installed on the tester's machine.
Hence, it might not be suitable to add Safari to this config.

However, I can create a separate Karma config for our MacOS GitHub workflow, and I'll add Safari for the tests in this workflow.

Reply:
Sure! Sounds good.

Reply:
Looks like we cannot use Safari for our CI checks, as it does not support "headless" mode.
Sources: [1](https://stackoverflow.com/questions/57687682/running-e2e-tests-in-headless-safari-using-nightwatch), [2](https://discussions.apple.com/thread/251837694)